components/base/kinematicbase/ptgKinematics.go

```diff
@@ -157,21 +159,28 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
 		linVel := r3.Vector{0, trajNode.LinVelMMPS, 0}
 		angVel := r3.Vector{0, 0, rdkutils.RadToDeg(trajNode.AngVelRPS)}
 
-		ptgk.logger.Debugf(
-			"setting velocity to linear %v angular %v and running velocity step for %s",
-			linVel,
-			angVel,
-			timestep,
-		)
-
-		err := ptgk.Base.SetVelocity(
-			ctx,
-			linVel,
-			angVel,
-			nil,
-		)
-		if err != nil {
-			return multierr.Combine(err, ptgk.Base.Stop(ctx, nil))
+		// This should call SetVelocity if:
+		// 1) this is the first iteration of the loop, or
+		// 2) either of the linear or angular velocities has changed
+		if i == 0 || !(linVel.ApproxEqual(lastLinVel) && angVel.ApproxEqual(lastAngVel)) {
```

Comment:
What are the units we are operating on here? Millimeters? 
If so I take it that that the `1e-16` epsilon that [ApproxEqual](https://github.com/golang/geo/blob/6adc56603217/r3/vector.go#L30) uses is what makes this semantically equivalent to an equality check. 

Is that right?

Reply:
You are correct regarding the equality check.

Units are mm per second for linVel, and degress per second for angVel.

I can add a comment specifying that if you like; the reason there isn't already is I considered the lines above to be self-documenting:
```
		linVel := r3.Vector{0, trajNode.LinVelMMPS, 0}
		angVel := r3.Vector{0, 0, rdkutils.RadToDeg(trajNode.AngVelRPS)}
```

Reply:
no need to comment this is good, just confirming my own understanding as I onboard.